R/autosyn.R

```diff
@@ -726,3 +753,75 @@ extract_ntpredictions.neuron <- function(x,
   }
 }
 
+# Get synapses as FlyWire annotations
+#' @export
+#' @rdname flywire_neurons_add_synapses
+flywire_synapse_annotations <- function(x,
+                                        file = NULL,
+                                        scale = 1/c(4,4,40), # from nm to voxel space
+                                        sample = NULL,
+                                        best = TRUE,
+                                        cleft.threshold = 30,
+                                        remove_autapses = TRUE,
+                                        local = NULL, # "/Volumes/GoogleDrive/Shared drives/hemibrain/fafbsynapses"
+                                        cloudvolume.url = NULL){
+  if(is.data.frame(x)||is.table(x)){
+    synapse.sample = x
+    if("cleft_scores"%in%colnames(x)){
+      synapse.sample <- synapse.sample %>%
+        dplyr::filter(.data$cleft_scores > cleft.threshold) %>%
+        dplyr::collect()
+    }
+  }else{
+    synapse.sample = flywire_ntpred(x,
+                                    cleft.threshold=cleft.threshold,
+                                    remove_autapses=remove_autapses,
+                                    local=local,
+                                    cloudvolume.url=cloudvolume.url)
+    synapse.sample[,c("x","y","z")] = synapse.sample[,c("pre_x","pre_y","pre_z")]
+  }
+  if(!is.null(sample)){
+    if(!is.integer(sample)){
```

Comment:
I recommend `checkmate::asInt(sample)` (the package is already in Imports) because otherwise `sample=1` will give an error because:
```
> is.integer(1)
[1] FALSE
```
because it is a double.

Reply:
Suggestion taken